File: 백준/Silver/15657. N과 M （8）/N과 M （8）.java
import java.io.*;
import java.util.*;

public class Main {	
	static int N, M;
	static int[] num, b;
	static StringBuilder sb = new StringBuilder();
	
	public static void main(String[] args) throws IOException  {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		N = Integer.parseInt(st.nextToken());
		M = Integer.parseInt(st.nextToken());
		
		num = new int[N];
		b = new int[M];
		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < N; i++) {
			num[i] = Integer.parseInt(st.nextToken());
		}
		
		Arrays.sort(num);
		
		comb(0, 0);
		
		System.out.println(sb);
		br.close();
	}
	private static void comb(int cnt, int start) {
		if(cnt == M) {
			sb.append(Arrays.toString(b).replace("[", "").replace(",", "").replace("]", "") + "\n");
			return;
		}
		for (int i = start; i < N; i++) {
			b[cnt] = num[i];
			comb(cnt+1, i);
		}
	}
}

File: 백준/Silver/15657. N과 M （8）/N과 M （8）.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
num = list(map(int, input().split()))

num.sort()

v = [[False] for _ in range(N)]
result = [[0] for _ in range(M)]

def comb(cnt, start) :
    if cnt == M :
        for i in range(M) :
            print(result[i], end=" ")
        print()
        return

    for i in range(start, N) :
        result[cnt] = num[i]
        comb(cnt+1, i)

comb(0, 0)